### DISCUSSION TITLE
Adding gradle plugins to a project

### DISCUSSION THREAD

**Discussion by pnewell** [skiptools/skip#73] [Q&A] [answered]
What is the correct way to add a plugin to my skip project? Normally this is added to the project-level build.gradle file, but I don't see any way to do that in skip and when I open the Android project in skip do not see a project-level build.gradle file at all...

**Comment by marcprux** [comment, marked as answer]
For a standard `skip init` generated project, the relevant Android project files are `Android/settings.gradle.kts` and `Android/app/build.gradle.kts`. You can add anything you want to those build configuration files, just don't break the parts that execute the Skip build. 

Note that if you are currently using the `groovy` variant of Gradle (as suggested by the `build.gradle` file name), you will need to translate it into the `kotlinscript` Gradle variant (in a `build.gradle.kts` file). 

FTR, your project structure should look something like this:

```plaintext
.
├── Android
│   ├── app
│   │   ├── build.gradle.kts
│   │   ├── proguard-rules.pro
│   │   └── src
│   │       └── main
│   │           ├── AndroidManifest.xml
│   │           └── kotlin
│   │               └── every
│   │                   └── thing
│   │                       └── Main.kt
│   ├── gradle
│   │   └── wrapper
│   │       └── gradle-wrapper.properties
│   ├── gradle.properties
│   └── settings.gradle.kts
├── Darwin
│   ├── Assets.xcassets
│   │   ├── AccentColor.colorset
│   │   └── Contents.json
│   ├── Entitlements.plist
│   ├── EveryThing.xcconfig
│   ├── EveryThing.xcodeproj
│   │   └── project.pbxproj
│   └── Sources
│       └── EveryThingAppMain.swift
├── Package.swift
├── README.md
├── Skip.env
└── Sources
    ├── EveryThing
    │   ├── ContentView.swift
    │   ├── Resources
    │   │   └── Localizable.xcstrings
    │   └── Skip
    │       └── skip.yml
    └── EveryThingApp
        └── EveryThingAppMain.swift

```
